File: 프로그래머스/1/42840. 모의고사/모의고사.py
def one_correct(idx,n):
    sub_list=[1,2,3,4,5]
    return sub_list[idx%5]==n

def two_correct(idx,n):
    sub_list=[2,1,2,3,2,4,2,5]    
    return sub_list[idx%8]==n

def three_correct(idx,n):    
    sub_list=[3,3,1,1,2,2,4,4,5,5]
    return sub_list[idx%10]==n

def solution(answers):
    supoza=[0,0,0]
    for idx,n in enumerate(answers):
        if one_correct(idx,n)==True:
            supoza[0]+=1
        if two_correct(idx,n)==True:
            supoza[1]+=1
        if three_correct(idx,n)==True:
            supoza[2]+=1
    max_count=max(supoza)
    answer=[]
    for idx,i in enumerate(supoza):
        if max_count==i:
            answer.append(idx+1)            
    
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> pattern1 = {1, 2, 3, 4, 5};
    vector<int> pattern2 = {2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> pattern3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    
    vector<int> scores(3, 0);
    
    for (int i = 0; i < answers.size(); i++) {
        if (answers[i] == pattern1[i % pattern1.size()]) scores[0]++;
        if (answers[i] == pattern2[i % pattern2.size()]) scores[1]++;
        if (answers[i] == pattern3[i % pattern3.size()]) scores[2]++;
    }
    
    int maxScore = max({scores[0], scores[1], scores[2]});
    
    vector<int> answer;
    for (int i = 0; i < 3; i++) {
        if (scores[i] == maxScore) answer.push_back(i + 1);
    }
    
    return answer;
}
